### PR TITLE
Support apply PVC template changes to StatefulSet

### PR DESCRIPTION
**Pull request by d-kuro**
refs: https://github.com/cybozu-go/moco/issues/265

~~Add design documentation for moco-controller to support volume size expansion~~

> **Note**
>
> Rewrote docs to focus on recreating the StatefulSet rather than extending volume

### CODE REVIEW

**Review by ymmt2005 — DISMISSED** [review]
LGTM

File: docs/designdoc/support_volume_expansion.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+kind: MySQLCluster
+status:
+  conditions:
+    - lastTransitionTime: "2022-05-01T16:29:24Z"
+      status: "False"
+      type: VolumeResized
+      message: "Validation failed: ..."
+```
+
+Set the `VolumeResized` condition's status to `True` if the PVC expansion was successful.
```
Comment (masa213f) [inline docs/designdoc/support_volume_expansion.md:78]: That fine with me.
Could you also export this status as a metric?

Comment (d-kuro) [reply docs/designdoc/support_volume_expansion.md:78]: I see, I came up with two options, which would you prefer?

1. Count metric, `moco_pvc_expansion_failed`
    * Count is incremented if PVC resizing fails
2. Gauge metric, `moco_pvc_expansion_status{states="True/False"}`
    * Aggregate the number of objects from the list of MySQLCluster's that currently exist with that status

Comment (masa213f) [reply docs/designdoc/support_volume_expansion.md:78]: Could you export this metrics for each MySQLCluster?
And also use the prefix of `moco_cluster_` as with other metrics.
https://github.com/cybozu-go/moco/blob/main/docs/metrics.md#mysql-clusters

How about use the following format?
Count metircs: `moco_cluster_volume_resized_total{status="True/False", name="mycluster", namespace="mynamesapce"}`

Comment (ymmt2005) [reply docs/designdoc/support_volume_expansion.md:78]: @masa213f 
As I told you elsewhere, moco-controller should re-create StatefulSet not only when the storage size is changed, but also when metadata (labels / annotations) is changed. Considering that, the metric name containing `volume_resized` is not appropriate.

Comment (ymmt2005) [reply docs/designdoc/support_volume_expansion.md:78]: @d-kuro Have you heard about that?

Comment (d-kuro) [reply docs/designdoc/support_volume_expansion.md:78]: I updated the docs to reflect the comments 🙏 :
https://github.com/cybozu-go/moco/pull/412/commits/e63a1d56f2c81684fc0c7492e43058c1538c4735

**Comment by d-kuro** [comment]
@ymmt2005 @masa213f 

I have rewritten the document based on conversations we had at our meetings.
Added metrics to notify errors, as moco-controller may endlessly handle errors that are not automatically recovered in some cases.
Please re-review :pray:
https://github.com/cybozu-go/moco/pull/412/commits/11475fa8d5bdad5689fc82cff3a5e0011a05b3c6

**Review by masa213f — APPROVED** [review]
LGTM. Great job! Thank you.

**Comment by donch** [comment]
Hello, 

Do you have any release date which will include this feature please ?

**Comment by masa213f** [comment]
@donch 
thank you for contacting. I will release it in the next week.

**Comment by masa213f** [comment]
Sorry, preparation of release has been delayed. I will release it next week. :(

**Pull request merged**